pkg/storage/engine.go

```diff
@@ -963,6 +964,11 @@ type Engine interface {
 	// All iterators created from a read-only engine are guaranteed to provide a
 	// consistent snapshot of the underlying engine. See the comment on the
 	// Reader interface and the Reader.ConsistentIterators method.
+	//
+	// NB: consistent iterators are not as of when NewReader is called. They are
+	// as of the time when the first iterator is created (often, this is when the
+	// first read happens if iterators are not used directly), or earlier if
```

Comment:
Looking for suggestion on how to word this. Iterators are always used, but I mean to say that they are often hidden behind things (e.g. `MVCCGet`), so it can be the "first read/access" from the user's perspective.